src/Console/Command/ObfuscateData.php

```diff
@@ -0,0 +1,305 @@
+            $this->removeDataProviderValues();
+            $this->overwriteSiteConfig();
+            $this->deleteUsers();
+            $this->addAdminUser();
+            $this->db->commit();
+        } else {
+            $this->info("This script will only run on test or staging deployments.\n\n");
+            exit;
+        }
+    }
+
+    protected function isThisAMultisiteInstall()
+    {
+        // @TODO: is this the correct way to check against multisite
+        return (config('multisite.domain'));
+    }
+
+    private function isTestEnvironment()
+    {
+        $cur_env = $this->getLaravel()->environment();
+        return (strtolower($cur_env) == 'test');
+    }
+
+    private function isStagingEnvironment()
+    {
+        $cur_env = $this->getLaravel()->environment();
+        return (strtolower($cur_env) == 'staging');
+    }
+
+    /* optional implementations...
+    private function randomizeLongitudeLatitude($boundaries)
+    {
+    }
+    */
+
+    /* TODO: would this be useful?
+    private function scrambleLongitudeLatitude($distanceFromOrigin = null)
+    {
+    }
+    */
+
+    private function overwriteContacts()
+    {
+        $this->info("Overwriting contact data...");
+        $faker = Faker\Factory::create();
+        $query = DB::select('contacts.id', 'contacts.type', 'contacts.contact')
+            ->distinct(true)
+            ->from('contacts');
+        $results = $query->execute($this->db);
+        $overwrittenCount = 0;
+        $resultsCount = count($results);
+        foreach ($results as $row) {
+            $contactEntity = $this->contactRepository->getEntity($row);
+            if ($row['type'] == 'phone') {
+                $fakePhone = $faker->e164PhoneNumber;
+                $contactEntity->setState(['contact'=> $fakePhone]);
+            } elseif ($row['type'] == 'email') {
+                $fakeEmail = $faker->safeEmail;
+                $contactEntity->setState(['contact'=> $fakeEmail]);
+            } elseif ($row['type'] == 'twitter') {
+                $fakeTwitter = '@fake'.preg_replace("/[^A-Za-z0-9]/", '', $faker->realText(15));
+                $contactEntity->setState(['contact'=> $fakeTwitter]);
+            } else {
+                $fakeText = preg_replace("/[^A-Za-z0-9]/", '', $faker->realText(20));
+                $contactEntity->setState(['contact'=> $fakeText]);
+            }
+            $overwrittenCount += $this->contactRepository->update($contactEntity);
+        }
+        if ($overwrittenCount == $resultsCount) {
+            $this->info("updated ".$overwrittenCount." records.\n");
+            return $overwrittenCount;
+        } else {
+            $this->info("failed to overwrite all records.\n");
+            return false;
+        }
+    }
+
+    private function overwritePostAuthors()
+    {
+        $this->info("Overwriting post author data...");
+        $faker = Faker\Factory::create();
+        $results = DB::select('posts.*')
```

Comment:
I know we couldn't decide on where this code should live before, but looking at it now I think we'd be better to move all the DB code into an `obfuscate` method on the repo, that just returns the number of records processed (or an exception if it fails).